app/src/components/login/Login.tsx

```diff
@@ -1,10 +1,112 @@
-import React from "react";
+import React, { useState } from "react";
+import Input from "../input";
+import isEmailValid from "../../utils/isEmailValid";
 import "./Login.css";
+import Button from "../button";
+import { Link } from "react-router-dom";
+import googleIcon from "../../images/googleicon.svg";
+import axios from "axios";
 
 function Login() {
+  const [email, setEmail] = useState("");
+  const [isEmailError, setIsEmailError] = useState(false);
+  const validateEmail = (currentEmail: string) => {
+    setIsEmailError(!isEmailValid(currentEmail));
+  };
+  const handleEmailOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value ? validateEmail(e.target.value) : setIsEmailError(false);
+    setEmail(e.target.value);
+  };
+  const handleSignin = async () => {
+    const errors =
+      (!isEmailValid(email) && setIsEmailError(true)) ||
+      validatePassword(password);
+    if (!errors) {
+      try {
+        const req = await axios({
+          method: "post",
+          url: "localhost:5000/auth/local",
+          data: {
+            email,
+            password,
+          },
+        });
+        if (req.status === 401) {
+          setPasswordErrorMessage("Invalid login!");
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    }
+  };
+  const [password, setPassword] = useState("");
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState("");
+  const validatePassword = (currentPassword: string) => {
+    let errors = [];
+    currentPassword.length < 8 && errors.push("at least 8 characters");
+    (!currentPassword.match(/[A-Z]/) || !currentPassword.match(/[a-z]/)) &&
+      errors.push("a mix of uppercase and lowercase");
+    setPasswordErrorMessage(
+      errors.length > 0
+        ? ["Password requires ", errors.join(" and ")].join("")
+        : ""
+    );
+    return errors.length > 0;
+  };
```

Comment:
Nice! Do you think it's a good idea to factor this out into a utility method?

Reply:
Yes good idea, we'll probably want to use it for the signup too